SimPPS/PPSPixelDigiProducer/src/RPixDummyROCSimulator.cc

```diff
@@ -0,0 +1,79 @@
+#include "SimPPS/PPSPixelDigiProducer/interface/RPixDummyROCSimulator.h"
+#include "Geometry/VeryForwardGeometry/interface/CTPPSPixelTopology.h"
+#include <vector>
+#include "TRandom.h"
+#include <iostream>
+
+RPixDummyROCSimulator::RPixDummyROCSimulator(const edm::ParameterSet &params, uint32_t det_id) : det_id_(det_id) {
+  threshold_ = params.getParameter<double>("RPixDummyROCThreshold");
+  electron_per_adc_ = params.getParameter<double>("RPixDummyROCElectronPerADC");
+  VcaltoElectronGain_ = params.getParameter<int>("VCaltoElectronGain");
+  VcaltoElectronOffset_ = params.getParameter<int>("VCaltoElectronOffset");
+  doSingleCalibration_ = params.getParameter<bool>("doSingleCalibration");
+  dead_pixel_probability_ = params.getParameter<double>("RPixDeadPixelProbability");
+  dead_pixels_simulation_on_ = params.getParameter<bool>("RPixDeadPixelSimulationOn");
+  pixels_no_ = CTPPSPixelTopology().detPixelNo();
+  verbosity_ = params.getParameter<int>("RPixVerbosity");
+  links_persistence_ = params.getParameter<bool>("CTPPSPixelDigiSimHitRelationsPersistence");
+}
+
+void RPixDummyROCSimulator::ConvertChargeToHits(
+    const std::map<unsigned short, double> &signals,
+    std::map<unsigned short, std::vector<std::pair<int, double> > > &theSignalProvenance,
+    std::vector<CTPPSPixelDigi> &output_digi,
+    std::vector<std::vector<std::pair<int, double> > > &output_digi_links,
+    const CTPPSPixelGainCalibrations *pcalibrations) {
+  for (std::map<unsigned short, double>::const_iterator i = signals.begin(); i != signals.end(); ++i) {
+    //one threshold per hybrid
+    unsigned short pixel_no = i->first;
+    if (verbosity_)
+      edm::LogInfo("RPixDummyROCSimulator") << "Dummy ROC adc and threshold : " << i->second << ", " << threshold_;
+    if (i->second > threshold_ && (!dead_pixels_simulation_on_ || dead_pixels_.find(pixel_no) == dead_pixels_.end())) {
+      float gain = 0;
+      float pedestal = 0;
+      int adc = 0;
+      uint32_t col = pixel_no / 160;
+      uint32_t row = pixel_no % 160;
+
+      const CTPPSPixelGainCalibration &DetCalibs = pcalibrations->getGainCalibration(det_id_);
+
+      // Avoid exception due to col > 103 in case of 2x2 plane. To be removed
+      if (col >= DetCalibs.getNCols())
+        continue;
+
+      if (doSingleCalibration_) {
+        adc = int(round(i->second / electron_per_adc_));
+      } else {
+        if (DetCalibs.getDetId() != 0) {
+          gain = DetCalibs.getGain(col, row) * 1800. / 260.;  // *highRangeCal/lowRangeCal
```

Comment:
@mundim , 1800, 260 and 255 better to define as const value like highRangeCal, lowRangeCal

Reply:
done